mart/utils/adapters.py

```diff
@@ -36,3 +38,23 @@ def __call__(self, *args, **kwargs):
         assert callable(function)
 
         return function(*args, **kwargs)
+
+
+class PartialInstanceWrapper:
+    """Make a partial class object callable."""
+
+    def __init__(self, partial: Callable, adapter):
+        """
+
+        Args:
+            partial (Callable): A partial of a class object.
+            adapter (CallableAdapter): An adapter that creates the `__call__` method.
+        """
+        self.partial = partial
+        self.adapter = adapter
```

Comment:
I would name this `wrapper` given the name of the class. I also think the only type requirement on this object that it is Callable.

Reply:
Fixed.